osu.Game/Screens/Play/ReplayDownloadButton.cs

```diff
@@ -0,0 +1,177 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Allocation;
+using osu.Framework.Graphics.Cursor;
+using osu.Framework.Graphics.Shapes;
+using osu.Framework.Graphics;
+using osu.Game.Graphics;
+using osu.Game.Graphics.Containers;
+using osu.Game.Online;
+using osu.Game.Scoring;
+using osuTK;
+using osu.Framework.Graphics.Sprites;
+using osu.Game.Online.API.Requests.Responses;
+using osu.Framework.Graphics.Effects;
+using osuTK.Graphics;
+using osu.Framework.Extensions.Color4Extensions;
+using osu.Framework.Graphics.Containers;
+
+namespace osu.Game.Screens.Play
+{
+    public class ReplayDownloadButton : DownloadTrackingComposite<ScoreInfo, ScoreManager>, IHasTooltip
+    {
+        private const int size = 40;
+
+        [Resolved(canBeNull: true)]
+        private OsuGame game { get; set; }
+
+        [Resolved]
+        private ScoreManager scores { get; set; }
+
+        [Resolved]
+        private OsuColour colours { get; set; }
```

Comment:
We could move both `OsuGame game` and `OsuColour colours` into the dependency loader as they are not used anywhere else in the code.